src/runtime/auto-prompt-manager.js

```diff
@@ -365,56 +388,59 @@ export class AutoPromptManager {
    *   dismissedPrompts: (?string|undefined),
    *   shouldShowAutoPrompt: (boolean|undefined),
    * }} params
-   * @return {!string|undefined}
+   * @return {!Promise<string|undefined>}
    */
   getAudienceActionPromptType_({
     article,
     autoPromptType,
     dismissedPrompts,
     shouldShowAutoPrompt,
   }) {
-    let potentialActions = article?.audienceActions?.actions || [];
-    potentialActions = potentialActions.filter((action) =>
+    const audienceActions = article?.audienceActions?.actions || [];
+
+    const filterBooleans = audienceActions.map((action) =>
```

Comment:
What about reading local storage in this function and pass an argument in `checkActionEligibility_`?

Reply:
I refactored the promises to be resolved prior to the eligibility check, and save the local storage fetches to `fetchedLocalStorage`. PTAL.

Reply:
Resolving given more recent discussion below.